ampd-handlers/src/bin/stellar/handler.rs

```diff
@@ -0,0 +1,578 @@
+use ampd_handlers::voting::{self, Error, PollEventData as _, VotingHandler};
+use ampd_sdk::event::event_handler::{EventHandler, SubscriptionParams};
+use ampd_sdk::grpc::client::EventHandlerClient;
+use async_trait::async_trait;
+use axelar_wasm_std::chain::ChainName;
+use axelar_wasm_std::hash::Hash;
+use axelar_wasm_std::msg_id::HexTxHashAndEventIndex;
+use axelar_wasm_std::voting::{PollId, Vote};
+use cosmrs::{AccountId, Any};
+use cosmwasm_std::HexBinary;
+use error_stack::{Report, ResultExt};
+use events::{try_from, AbciEventTypeFilter, Event, EventType};
+use serde::Deserialize;
+use serde_with::{serde_as, DisplayFromStr};
+use stellar_xdr::curr::ScAddress;
+use typed_builder::TypedBuilder;
+
+pub type Result<T> = error_stack::Result<T, Error>;
+
+#[serde_as]
+#[derive(Clone, Debug, Deserialize)]
+#[try_from("wasm-messages_poll_started")]
+pub struct MessagesPollStarted {
+    poll_id: PollId,
+    source_chain: ChainName,
+    #[serde_as(as = "DisplayFromStr")]
+    source_gateway_address: ScAddress,
+    confirmation_height: u64,
+    expires_at: u64,
+    messages: Vec<Message>,
+    participants: Vec<AccountId>,
+}
+
+#[serde_as]
+#[derive(Clone, Debug, Deserialize)]
+#[try_from("wasm-verifier_set_poll_started")]
+pub struct VerifierSetPollStarted {
+    verifier_set: VerifierSetConfirmation,
+    poll_id: PollId,
+    source_chain: ChainName,
+    #[serde_as(as = "DisplayFromStr")]
+    source_gateway_address: ScAddress,
+    expires_at: u64,
+    confirmation_height: u64,
+    participants: Vec<AccountId>,
+}
+
+#[derive(Clone, Debug)]
+pub enum PollStartedEvent {
+    Messages(MessagesPollStarted),
+    VerifierSet(VerifierSetPollStarted),
+}
+
+impl TryFrom<Event> for PollStartedEvent {
+    type Error = Report<events::Error>;
+
+    fn try_from(event: Event) -> std::result::Result<Self, Self::Error> {
+        if let Ok(event) = MessagesPollStarted::try_from(event.clone()) {
+            Ok(PollStartedEvent::Messages(event))
+        } else if let Ok(event) = VerifierSetPollStarted::try_from(event.clone()) {
+            Ok(PollStartedEvent::VerifierSet(event))
+        } else {
+            Err(Report::new(events::Error::EventTypeMismatch(format!(
+                "{}/{}",
+                MessagesPollStarted::event_type(),
+                VerifierSetPollStarted::event_type()
+            )))
+            .attach_printable(format!("{{ event = {event:?} }}")))
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum PollEventData {
+    Message(Message),
+    VerifierSet(VerifierSetConfirmation),
+}
+
+impl voting::PollEventData for PollEventData {
+    type Digest = Hash;
+    type MessageId = HexTxHashAndEventIndex;
+    type ChainAddress = ScAddress;
+    type Receipt = TxResponse;
+
+    fn tx_hash(&self) -> Self::Digest {
+        match self {
+            PollEventData::Message(message) => message.message_id.tx_hash,
+            PollEventData::VerifierSet(verifier_set) => verifier_set.message_id.tx_hash,
```

Comment:
comparing to the old stellar handlers I noticed they used `tx_hash_as_hex_no_prefix` instead of `tx_hash`, should those functions be used here instead?